vtr_flow/arch/COFFE_22nm/k6FracN10LB_mem20K_complexDSP_customSB_22nm.clustered.denser.xml

```diff
@@ -0,0 +1,3159 @@
+    The architecture also has a 20Kb memory blocks (or M20k or BRAM) that has true and simple dual port modes. 
+    In simple dual port mode the memory can be configured in the following modes: 512x40, 1024x20 and 2048x10,
+    while in true dual port mode it can be configured as: 1024x20 and 2048x10.
+
+    The BRAM has registered inputs and outputs. See details on how the delays for this block were 
+    obtained, in the comments before the specification of the BRAM primitive, towards the end of this file.
+
+    The BRAM doesn't have an input crossbar. Adding an input crossbar was leading to a 
+    seg fault in VPR, likely because of https://github.com/verilog-to-routing/vtr-verilog-to-routing/issues/1475
+
+    Once this is fixed, an input crossbar can be added. The input crossbar delay from COFFE was: 29.47ps
+
+    ##############################
+    Routing/Interconnect
+    ##############################
+    The routing channel width is 300. Note that the channel width isn't specified directly in this arch file. 
+    Switch pattern calculations assume that value. During experiments, channel width can be specified using 
+    the command line switch `route_chan_width`.
+    The architecture uses unidirectional routing with wire segments of length 4 (260 out of 300 wires) and 
+    length 16 (40 out of 300 wires). The length 16 wires do not directly connect to block pins and are only 
+    accessible from the length 4 wires. Switches appear after every 4 blocks on the length 16 wires. 
+    The switch blocks use a custom switching pattern based on the Stratix-IV-like architecture used in the 
+    Titan flow [5]. 
+
+    ##############################
+    I/Os
+    ##############################
+    I/O pads are arranged along the perimeter of the FPGA. No area values provided for the I/Os.
+
+    ##############################
+    What makes this architecture Agilex-like?
+    ##############################
+    1. The DSP slice supports lower precision modes - int8 (actually 9x9) and 16-bit floating point
+       (IEEE half-precision and bfloat16).
+    2. DSPs are chained in vertical direction (currently only chainin-chainout are supported; 
+       in the future, scanin-scanout connections will be added as well)
+    3. Instead of having wires of just one length in the entire FPGA, there are wire segments of L=4 
+       and L=16. And a custom switch pattern (not a standard wilton switch) is used. 
+
+    ##############################
+    What makes this architecture different from Agilex and Stratix10?
+    ##############################
+    1. There are no registers on the interconnect/routing wires in this architecture. That is a main
+       feature in the Stratix10 and Agilex families of Intel FPGA (it's called HyperFlex by Intel)
+    2. The architecture doesn't have sectors. All blocks are laid out in columns on the entire chip.
+    3. The IOs are on the perimter, instead of being arranged in columns.
+
```

Comment:
Routing architecture (wire lengths, switch topologies) are also very different from Agilex; should mention that (and in other arch files).